homeassistant/components/overkiz/config_flow.py

```diff
@@ -1,78 +1,221 @@
+            gateways = await client.get_gateways()
+
+            gateway_id = ""
+            for gateway in gateways:
+                # Overkiz can return multiple gateways, but we only can generate a token
+                # for the main gateway.
+                if is_overkiz_gateway(gateway.id):
+                    gateway_id = gateway.id
+
+            developer_mode = await client.get_setup_option(
+                f"developerMode-{gateway_id}"
+            )
+
+            if developer_mode is None:
+                raise DeveloperModeDisabled
+
+            token = await client.generate_local_token(gateway_id)
+            uuid = await client.activate_local_token(
+                gateway_id=gateway_id, token=token, label="Home Assistant/local"
+            )
+
+            # Verify SSL blocked by https://github.com/Somfy-Developer/Somfy-TaHoma-Developer-Mode/issues/5
+            # Somfy (self-signed) SSL cert uses the wrong common name
+            session = async_create_clientsession(self.hass, verify_ssl=False)
+
+            # Local API
+            local_client = OverkizClient(
+                username="",
+                password="",
+                token=token,
+                session=session,
+                server=generate_local_server(host=user_input[CONF_HOST]),
+            )
+
+            try:
+                await local_client.login()
+            except Exception as exception:  # pylint: disable=broad-except
+                try:
+                    # Remove local token when login is not successful
+                    await client.delete_local_token(gateway_id, uuid)
+                except NotSuchTokenException:
+                    LOGGER.debug("Token is invalid / has been already removed")
+
+                raise exception
 
-        await client.login(register_event_listener=False)
+            user_input[CONF_TOKEN] = token
+            user_input[CONF_TOKEN_UUID] = uuid
 
-        # Set first gateway id as unique id
+        else:
+            server = SUPPORTED_SERVERS[user_input[CONF_SERVER]]
+
+            session = async_create_clientsession(self.hass)
+            client = OverkizClient(
+                username=username, password=password, server=server, session=session
+            )
+
+            await client.login(register_event_listener=False)
+
+        # Set main gateway id as unique id
         if gateways := await client.get_gateways():
-            gateway_id = gateways[0].id
-            await self.async_set_unique_id(gateway_id)
+            for gateway in gateways:
+                if is_overkiz_gateway(gateway.id):
+                    gateway_id = gateway.id
+                    await self.async_set_unique_id(gateway_id)
+
+        return user_input
 
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle the initial step via config flow."""
-        errors = {}
+        errors: dict[str, str] = {}
```

Comment:
```suggestion
```

Reply:
Fixed in 89981d36b78fc1a8049bee7e6eff4a5926a700a7.